Dynalite Integration

ziv1234 opened the pull request:
## Breaking Change:

Not a breaking change. Only adds a new integration

<!-- What is breaking and why we have to break it. Remove this section only if it was NOT a breaking change. -->

## Description:

I used Troy Kelly's library and component as a starting point, added several features to them, and am now able to control my full house which is based on Dynalite. I know that no other open-source system supports it, so will be glad if it can go into the mainstream, so others can use it as well. Will also be happy to make fixes to issues that come.
Since I am new to open-source development, wanted to understand what is needed to get this merged. One thing that is clearly lacking today is documentation, since this is something that is only worthwhile investing in if others will read it, so if you tell me that this should be integrated into HA, I will quickly document it.
I tried to follow the guidelines in the dev docs, but I am sure that there are some things that I missed. If this is interesting to add, please let me know if there is anything other than the documentation that needs fixing and i will take care of it quickly.

Best,
Ziv


**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#<home-assistant.io PR number goes here>

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

ziv1234 committed:
Initial work

ziv1234 committed:
Initial work

ziv1234 committed:
Merge remote-tracking branch 'upstream/dev' into dev

ziv1234 committed:
Mostly doc changes and fixing enums

ziv1234 committed:
Mostly doc changes and fixing enums

ziv1234 committed:
Merge remote-tracking branch 'upstream/dev' into dev

homeassistant commented:
Hi @ziv1234, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#27621).

Once you do that we will be able to review and accept this pull request.

Thanks!

danielperna84 committed:
Update pyhomematic to 0.1.61 (#27620)

homeassistant committed:
[ci skip] Translation update

jjlawren committed:
Rewrite Plex tests (#27624)

Michsior14 committed:
fix: exception after kaiterra api call timeout (#27622)

stevendlander committed:
Issue #27288 Moving imports to top for tesla component (#27618)

Mofeywalker committed:
add content for services.yaml for ccomponent stream (#27610)

rishatik92 committed:
Update yandex transport after api change (#27591)

* yandex maps api changed ("threads" in "Transport" added), ya_ma=>0.3.8 bug_fixed

* Update homeassistant/components/yandex_transport/sensor.py

Co-Authored-By: Paulus Schoutsen <paulus@home-assistant.io>

* additional fix

* reformat

* fix mistake

stevendlander committed:
Issue #27288 Move imports to top for FFMPEG (#27613)

exxamalte committed:
move imports to top-level (#27630)

pvizeli committed:
Update azure-pipelines-wheels.yml for Azure Pipelines

Michsior14 committed:
move imports in synology_srm component (#27603)

Bouni committed:
moved imports to top level (#27632)

Bouni committed:
moved imports to top level (#27634)

exxamalte committed:
Move imports in darksky component (#27633)

* move imports to top-level

* modify patch path

* removed unused mocks and patches

crazyfx1 committed:
Fix temperature and heating mode (#27604)

Bouni committed:
moved imports to top level (#27640)

OttoWinter committed:
ESPHome Fix intermediary state published (#27638)

Fixes https://github.com/esphome/issues/issues/426

Bouni committed:
moved imports to top level (#27641)

delphiki committed:
Fix html5 notification documentation url (#27636)

andrewsayre commented:
Thank you for this work, but there's too much going on in this single PR.  Please limit new integrations to a single platform only and strip out the features that aren't critical for it's operation.  This makes it easier for us to review and get you up to speed with the architecture and our coding style.  It will make future PRs must faster!

Please see the checklist for more info: https://developers.home-assistant.io/docs/en/creating_component_code_review.html#5-make-your-pull-request-as-small-as-possible

javicalle committed:
Apply isort on rfxtrx classes (#27615)

* Move imports in rfxtrx component

* Apply isort on rfxtrx files

* Update test_switch.py

OttoWinter committed:
Fix ESPHome climate preset mode refactor (#27637)

Fixes https://github.com/home-assistant/home-assistant/issues/25613

stevendlander committed:
Refactor imports for tensorflow (#27617)

* Refactoring imports for tensorflow

* Removing whitespace spaces on blank line 110

* Moving tensorflow to try/except block

* Fixed black formatting

* Refactoring try/except to if/else

Bouni committed:
Move imports in bluesound component (#27502)

* moved imports to top level

* changed import order

* changed import order

emontnemery committed:
Fix ZHA regressions caused by "Support async validation of device trigger" (#27401)

* Revert "Support async validation of device trigger (#27333)"

This reverts commit fdf4f398a79e775e11dc9fdd391a8b53f7b773c5.

* Revert only ZHA changes

* Fix whitespace

* Restore ZHA changes but add check to make sure ZHA is loaded

* Address review comment

* Remove additional check

ziv1234 commented:
Thanks. I am trying to think what I can take out. This is built similarly to the Hue lights component, where there is one bridge that connects to many devices. The devices I have there are pretty much empty shells that take all the logic from the library that handle the devices. This is implemented in light.py, switch,py, and cover.py for the different devices that can be behind a Dynalite hub. Many of the functions (e.g. name, unique_id, etc.) are common, so they are all inheriting from a dynalite_base.py for the common methods and attributes.

The bridge is quite straightforward. There is some logic there for assigning the entities to areas, which I can take out for the first PR, as it is not mandatory for the platform.

The other relatively large component is __init__.py. The reason for this is that there are many parameters for the Dynalite library, as the Dynalite systems have absolutely no auto-discover features. Almost none of this logic is used in the component, but it is needed for the vol schemas. Is it ok if I import the schema from the Dynalite library? This would significantly reduce the amount of code.

Another thing that I can do is to reduce the subplatforms (light, switch, cover) and start with only one (probably light). It will have limited functionality until everything is implemented, but can break the review cycle.

Will pull together something with all of this and send a new pull request.

Best,
Ziv

From: Andrew Sayre <notifications@github.com>
Sent: Monday, October 14, 2019 6:36 PM
To: home-assistant/home-assistant <home-assistant@noreply.github.com>
Cc: Ziv <zivk@hotmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [home-assistant/home-assistant] Dynalite Integration (#27621)


Thank you for this work, but there's too much going on in this single PR. Please limit new integrations to a single platform only and strip out the features that aren't critical for it's operation. This makes it easier for us to review and get you up to speed with the architecture and our coding style. It will make future PRs must faster!

Please see the checklist for more info: https://developers.home-assistant.io/docs/en/creating_component_code_review.html#5-make-your-pull-request-as-small-as-possible

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/home-assistant/home-assistant/pull/27621?email_source=notifications&email_token=AD5UNSYB3BOCSNQJ4DPRP4TQOSGXLA5CNFSM4JAIXR5KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBFHQPI#issuecomment-541751357>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD5UNSYXLG4GMSTYSLTP3UTQOSGXLANCNFSM4JAIXR5A>.

Michsior14 committed:
move imports in squeezebox component (#27650)

scop committed:
Typing misc fixes (#27543)

* Make async_get_conditions return type hint more specific

* Exclude script/scaffold/templates/ from pre-commit mypy

bramkragten committed:
Updated frontend to 20191014.0 (#27661)

ziv1234 committed:
changed to make it minimal

ziv1234 committed:
changed to make it minimal

ziv1234 committed:
ran script

ziv1234 committed:
ran script

javicalle committed:
Move imports in panasonic_viera component (#27665)

Kane610 committed:
Improve discovery title (#27664)

javicalle committed:
Move imports in panasonic_bluray component (#27658)

ochlocracy committed:
Update Unlock directive for Alexa LockController (#27653)

* Update the Alexa.LockController Unlock directive to include the lockState property in the context of the response.

* Added Test for Alexa.LockController Unlock directive to include the lockState property in the context of the response.

Michsior14 committed:
move imports in statsd component (#27649)

Michsior14 committed:
Move imports in steam_online component (#27648)

* move imports in steam_online component

* fix: import reassigment

Michsior14 committed:
move imports in stream component (#27647)

Michsior14 committed:
Move imports in switchmate component (#27646)

* move imports in switchmate component

* fix: bring back pylint ignore line

ziv1234 committed:
pylint fixes

ziv1234 committed:
pylint fixes

ziv1234 commented:
I believe i made it quite minimal now. is this better?

balloob committed:
Remove direct authentication via trusted networks or API password (#27656)

* Remove direct authentication via trusted networks and API password

* Fix tests

caronc committed:
Add Apprise notification integration (#26868)

* Added apprise notification component

* flake-8 fixes; black formatting + import merged to 1 line

* pylint issues resolved

* added github name to manifest.json

* import moved to top as per code review request

* manifest formatting to avoid failing ci

* .coveragerc updated to include apprise

* removed block for written tests

* more test coverage

* formatting as per code review

* tests converted to async style as per code review

* increased coverage

* bumped version of apprise to 0.8.1

* test that mocked entries are called

* added tests for hass.service loading

* support tags for those who identify the TARGET option

* renamed variable as per code review

* 'assert not' used instead of 'is False'

* added period (in case linter isn't happy)

homeassistant committed:
[ci skip] Translation update

scop committed:
Deprecate Python 3.6 support, 3.8.0 is out (#27680)

Bouni committed:
moved imports to top level (#27682)

Bouni committed:
moved imports to top level (#27678)

Bouni committed:
moved imports to top level (#27675)

Bouni committed:
moved imports to top level (#27677)

bwarden committed:
Handle marker attrs that may not exist (#27519)

marker-high-levels and marker-low-levels may not exist in printer
attributes returned by CUPS, so we'll use .get() to avoid this and
default to None:
KeyError: 'marker-high-levels'

Fixes #27518

SukramJ committed:
Improve code coverage for HomematicIP Cloud (#27606)

* Improve tests for HomematicIP Cloud

* create fixtures remove decorators

* removed further decorators

* remove  last decorator

* improve exception handling

* removed not required coroutine

* use the correct place for mock

Bouni committed:
Move imports in decora component (#27645)

* moved imports to top level

* replaced importlib with standard import

* fix for Unable to import 'decora' error

Bouni committed:
moved imports to top level (#27683)

Quentame committed:
Move imports in yeelight + yeelightsunflower component (#27388)

* Move imports in yeelight + yeelightsunflower component

* Fix pylint

* Fix pylint (again)

Bouni committed:
Move imports in bme680 component (#27506)

* moved imports to top level

* fixed pylint error

* moved imports to top level

* fixed import error

ratsept committed:
Fix missing strings in soma config flow (#27689)

Bouni committed:
moved imports to top level (#27695)

luca-angemi committed:
Add battery status in owntracks (#27686)

* Add battery status in owntracks

* Remove trailing whitespaces

Bouni committed:
moved imports to top level (#27693)

AaronDavidSchneider committed:
Update fritzconnection requirement to 0.8.4 (#27698)

* update fritzconnection requirement

* update requierements for other components and requierements_all

springstan committed:
Moved imports to top-level in spotify integration (#27703)

Bouni committed:
moved imports to top level (#27679)

quthla committed:
Bump PyMata to 2.20 (#27431)

* Bump PyMata to 2.20

* Bump PyMata to 2.20

abstrakct committed:
Add improved scene support to input_text (#27687)

* Add improved scene support for input_text.

* Add tests for reproducing input_text states.

* Add some comments.

ziv1234 committed:
unit tests

ziv1234 committed:
unit tests

ziv1234 committed:
added exception for invalid host

ziv1234 committed:
added exception for invalid host

ziv1234 committed:
test invalid host

ziv1234 committed:
test invalid host

ziv1234 committed:
unit test

ziv1234 committed:
unit test

ziv1234 committed:
unit tests

ziv1234 committed:
unit tests

balloob committed:
Fix config imports (#27669)

* Fix config imports

* Remove old migration

* Remove migrate tests

abstrakct committed:
Add improved scene support for input_select (#27697)

* Add improved scene support for input_select

* Add tests for reproducing input_select states.

* Add some comments.

* Add support for set_options

Allows defining the options for an input_select in a scene.

* Add tests for set_options in test_reproduce_state

* Execute for real instead of mock execution.

Michsior14 committed:
move imports in sony_projector component (#27718)

Michsior14 committed:
move imports in speedtestdotnet component (#27716)

Michsior14 committed:
move imports in spotcrime component (#27715)

springstan committed:
Moved imports to top-level in fritzbox_callmonitor component (#27705)

Bouni committed:
moved imports to top level (#27696)

ziv1234 committed:
unit tests

ziv1234 committed:
unit tests

homeassistant committed:
[ci skip] Translation update

ziv1234 committed:
unit tests - bridge almost done

ziv1234 committed:
unit tests - bridge almost done

ziv1234 committed:
unit test

ziv1234 committed:
unit test

ziv1234 committed:
unit tests

ziv1234 committed:
unit tests

bvlaicu committed:
New sensor platform integration for Orange and Rockland Utility smart energy meter (#27571)

* New sensor platform integration for Orange and Rockland Utility smart energy meter

* New sensor platform integration for Orange and Rockland Utility smart energy meter

* bumped the oru py version to 0.1.9

* Added PLATFORM_SCHEMA
Adde unique_id property
Changed logger level from info to debug when printing the updated sensor value
Set the SCAN_INTERVAL to 15 mins
Added exception handling durin init when creating the oru meter instance

* Various fixes base on the PR review
+ Added SCAN_INTERVAL for 15 mins

* fixed path to documentation

Bouni committed:
Move imports in device_tracker component (#27676)

* moved imports to top level

* sorted imports using isort

zxdavb committed:
Add evohome high_precision temperatures (#27513)

* add high_precision (current) temperatures 
* bump client to use aiohttp for v1 client
* token saving now event-driven rather than scheduled
* protection against invalid tokens that cause issues
* tweak error message

ziv1234 committed:
unit tests

ziv1234 committed:
unit tests

ziv1234 committed:
unit tests

ziv1234 committed:
unit tests

tulindo committed:
Move imports in Kodi component (#27728)

* Move imports for Kodi component

* Removed empty line ad requested by review

bieniu committed:
Add sensor platform to Airly integration (#27717)

* Add sesnor.py file

* Move AirlyData to __init__

* Cleaning

* Update .coveragerc file

* Sort consts

* Sort imports

* Remove icons from sensors with device_class

cgtobi committed:
Bump pyatmo version to 2.3.2 (#27731)

* Bump pyatmo version to 2.3.2

* Add reachable attribute

* Add reachable attribute

ziv1234 committed:
unit test for light.py

ziv1234 committed:
added test_light

ziv1234 committed:
Merge branch 'dev' of https://github.com/ziv1234/home-assistant into dev

ziv1234 committed:
simplified config_flow and unit tests

ziv1234 committed:
fixes for pylint cleanup

ziv1234 committed:
added tests

ziv1234 committed:
added __init__.py as empty

ziv1234 commented:
added unit tests and now it fully goes through the CI. I know that the documentation is missing, but wanted to know if this is something that you are likely to accept since I wouldn't want to spend a lot of time on good documentation that no one will read :)

Of course, if there is anything else that I can do to help, please let me know

balloob commented:
It seems like you have some commits in this branch that are not yours, please clean it up.

ziv1234 commented:
I pulled from the upstream branch, so these are all commits that were already accepted. I followed the “keeping up with reality” section in the docs.

Do you want me to create a new, clean fork? Is there any other way to cancel these pulled commits?

From: Paulus Schoutsen <notifications@github.com>
Sent: Friday, October 18, 2019 3:55 AM
To: home-assistant/home-assistant <home-assistant@noreply.github.com>
Cc: Ziv <zivk@hotmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [home-assistant/home-assistant] Dynalite Integration (#27621)


It seems like you have some commits in this branch that are not yours, please clean it up.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/home-assistant/home-assistant/pull/27621?email_source=notifications&email_token=AD5UNS4IQ6L3HARI526UWTDQPECOFA5CNFSM4JAIXR5KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBSB3BY#issuecomment-543432071>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD5UNS5ZN6H3TJHY5IHGLTDQPECOFANCNFSM4JAIXR5A>.

ziv1234 commented:
I created a new branch and a new pull request that is now clean. Will not pull more from upstream…

From: Paulus Schoutsen <notifications@github.com>
Sent: Friday, October 18, 2019 3:55 AM
To: home-assistant/home-assistant <home-assistant@noreply.github.com>
Cc: Ziv <zivk@hotmail.com>; Mention <mention@noreply.github.com>
Subject: Re: [home-assistant/home-assistant] Dynalite Integration (#27621)


It seems like you have some commits in this branch that are not yours, please clean it up.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/home-assistant/home-assistant/pull/27621?email_source=notifications&email_token=AD5UNS4IQ6L3HARI526UWTDQPECOFA5CNFSM4JAIXR5KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEBSB3BY#issuecomment-543432071>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AD5UNS5ZN6H3TJHY5IHGLTDQPECOFANCNFSM4JAIXR5A>.

ziv1234 commented:
closed since there is an updated pull request, 27841
